src/field.js

```diff
@@ -198,26 +183,76 @@ function Field(name, id, type, rule, extend, options, comment) {
 }
 
 /**
- * Determines whether this field is packed. Only relevant when repeated and working with proto2.
+ * Determines whether this field is required.
+ * @name Field#required
+ * @type {boolean}
+ * @readonly
+ */
+Object.defineProperty(Field.prototype, "required", {
+    get: function() {
+        return this._features.field_presence === "LEGACY_REQUIRED";
```

Comment:
Seems that this is the root of the issue

Reply:
See above comment, this is behavior preserving

Reply:
And yet the protobuf output between the previous version and this version is different. Now all fields are required which was NOT the case before